iceoryx_posh/test/moduletests/test_roudi_cmd_line_parser.cpp

```diff
@@ -349,83 +353,119 @@ TEST_F(CmdLineParser_test, CompatibilityLevelOptionsLeadToCorrectCompatibilityLe
+    CmdLineParser sut;
+    auto result = sut.parse(NUMBER_OF_ARGS, args);
 
-    EXPECT_THAT(result.has_error(), Eq(false));
-    EXPECT_THAT(result.value().uniqueRouDiId.has_value(), Eq(true));
-    EXPECT_THAT(result.value().uniqueRouDiId.value(), Eq(4242));
+    ASSERT_FALSE(result.has_error());
+    ASSERT_TRUE(result.value().uniqueRouDiId.has_value());
+    EXPECT_EQ(result.value().uniqueRouDiId.value(), 4242);
 }
 
 TEST_F(CmdLineParser_test, UniqueIdShortOptionLeadsToCorrectUniqueId)
 {
-    constexpr uint8_t numberOfArgs{3U};
-    char* args[numberOfArgs];
+    constexpr uint8_t NUMBER_OF_ARGS{3U};
+    char* args[NUMBER_OF_ARGS];
     char appName[] = "./foo";
     char option[] = "-u";
     char value[] = "4242";
     args[0] = &appName[0];
     args[1] = &option[0];
     args[2] = &value[0];
 
-    iox::config::CmdLineParser sut;
-    auto result = sut.parse(numberOfArgs, args);
+    CmdLineParser sut;
+    auto result = sut.parse(NUMBER_OF_ARGS, args);
 
-    EXPECT_THAT(result.has_error(), Eq(false));
-    EXPECT_THAT(result.value().uniqueRouDiId.has_value(), Eq(true));
-    EXPECT_THAT(result.value().uniqueRouDiId.value(), Eq(4242));
+    ASSERT_FALSE(result.has_error());
+    ASSERT_TRUE(result.value().uniqueRouDiId.has_value());
+    EXPECT_EQ(result.value().uniqueRouDiId.value(), 4242);
 }
 
 TEST_F(CmdLineParser_test, OutOfBoundsUniqueIdOptionLeadsToProgrammNotRunning)
 {
-    constexpr uint8_t numberOfArgs{3U};
-    char* args[numberOfArgs];
+    constexpr uint8_t NUMBER_OF_ARGS{3U};
+    char* args[NUMBER_OF_ARGS];
     char appName[] = "./foo";
     char option[] = "-u";
     char wrongValue[] = "65536"; // MAX_ROUDI_ID + 1
     args[0] = &appName[0];
     args[1] = &option[0];
     args[2] = &wrongValue[0];
 
-    iox::config::CmdLineParser sut;
-    auto result = sut.parse(numberOfArgs, args);
+    CmdLineParser sut;
+    auto result = sut.parse(NUMBER_OF_ARGS, args);
 
-    EXPECT_THAT(result.has_error(), Eq(false));
-    EXPECT_THAT(result.value().run, Eq(false));
+    ASSERT_FALSE(result.has_error());
+    EXPECT_FALSE(result.value().run);
 }
+
+TEST_F(CmdLineParser_test, CmdLineParsingModeEqualToOneHandlesOnlyTheFirstOption)
+{
+    constexpr uint8_t NUMBER_OF_ARGS{5U};
+    char* args[NUMBER_OF_ARGS];
+    char appName[] = "./foo";
+    char uniqueIdOption[] = "-u";
+    char idValue[] = "4242";
+    char killOption[] = "-k";
+    char killValue[] = "42";
+    args[0] = &appName[0];
+    args[1] = &uniqueIdOption[0];
+    args[2] = &idValue[0];
+    args[3] = &killOption[0];
+    args[4] = &killValue[0];
+
+    CmdLineParser sut;
+    auto result = sut.parse(NUMBER_OF_ARGS, args, CmdLineParser::CmdLineArgumentParsingMode::ONE);
+
+    ASSERT_FALSE(result.has_error());
+    ASSERT_TRUE(result.value().uniqueRouDiId.has_value());
+    EXPECT_EQ(result.value().uniqueRouDiId.value(), 4242);
+    EXPECT_EQ(result.value().processKillDelay, Duration::fromSeconds(45)); // default value for kill delay
```

Comment:
Please use ``iox::roudi::PROCESS_DEFAULT_KILL_DELAY`` instead of ``Duration::...``

Reply:
Looks like it's done.